src/pip/_internal/operations/install/wheel.py

```diff
@@ -96,13 +114,13 @@ def csv_io_kwargs(mode):
 
 
 def fix_script(path):
-    # type: (text_type) -> Optional[bool]
+    # type: (text_type) -> bool
     """Replace #!python with #!/path/to/python
     Return True if file was changed.
     """
     # XXX RECORD hashes will need to be updated
     if not os.path.isfile(path):
-        return None
+        return False
```

Comment:
When would this happen? Should we return something else (like `None` as in the previous implementation) so the call site can fix the RECORD entry?

Reply:
Only in race conditions, like another process acting on the same environment and removing/replacing the file out from under us. We don't guard against that, so I can switch this to `assert os.path.isfile(path)`.

Reply:
Fixed in 50f6f8d6.